examples/handlers/multipart/src/main.rs

```diff
@@ -0,0 +1,125 @@
+            Some(ct[idx + BOUNDARY.len()..].to_string())
+        })
+        .unwrap();
+
+    let f = Body::take_from(&mut state)
+        .concat2()
+        .then(|full_body| match full_body {
+            Ok(valid_body) => {
+                let mut m = Multipart::with_body(Cursor::new(valid_body), boundary);
+                match m.read_entry() {
+                    Ok(Some(mut field)) => {
+                        let mut data = Vec::new();
+                        field.data.read_to_end(&mut data).expect("can't read");
+                        let res_result = String::from_utf8(data);
+                        let mut res_body;
+                        match res_result {
+                            Ok(r) => res_body = r.to_string(),
+                            Err(e) => res_body = format!("{:?}", e),
+                        }
+                        let res =
+                            create_response(&state, StatusCode::OK, mime::TEXT_PLAIN, res_body);
+                        future::ok((state, res))
+                    }
+                    Ok(None) => {
+                        let res = create_response(
+                            &state,
+                            StatusCode::OK,
+                            mime::TEXT_PLAIN,
+                            "can't read".to_string(),
+                        );
+                        future::ok((state, res))
+                    }
+                    Err(e) => {
+                        let res = create_response(
+                            &state,
+                            StatusCode::OK,
+                            mime::TEXT_PLAIN,
+                            format!("{:?}", e),
+                        );
+                        future::ok((state, res))
+                    }
+                }
+            }
+            Err(e) => future::err((state, e.into_handler_error())),
+        });
+    Box::new(f)
+}
+
+/// Create a `Router`
+fn router() -> Router {
+    build_simple_router(|route| {
+        route.post("/").to(form_handler);
+    })
+}
+
+/// Start a server and use a `Router` to dispatch requests
+pub fn main() {
+    let addr = "127.0.0.1:7878";
+    println!("Listening for requests at http://{}", addr);
+    gotham::start(addr, router())
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use gotham::test::TestServer;
+
+    #[test]
+    fn form_request() {
+        let boundary = "--abcdef1234--";
+        let body = format!(
+            "\
+             --{0}\r\n\
+             content-disposition: form-data; name=\"foo\"\r\n\r\n\
+             bar\r\n\
+             --{0}--\r\n\
+             ",
+            boundary
+        );
+
+        let test_server = TestServer::new(router()).unwrap();
+        let response = test_server.client()
+            .post("http://localhost", body, mime::MULTIPART_FORM_DATA)
+            .perform()
+            .unwrap();
+        // let header_map = request.headers_mut();
```

Comment:
Here is how you add a header to your request:

```
        let test_server = TestServer::new(router()).unwrap();
        let client = test_server.client();
        let mut request = client.post("http://localhost", body, mime::MULTIPART_FORM_DATA);

        let headers = request.headers_mut();
        let h = format!("multipart/form-data; boundary={}", boundary);
        headers.insert(CONTENT_TYPE, HeaderValue::from_str(h.as_str()).unwrap());
        let response = request.perform().unwrap();
```

Reply:
@colinbankier  Thanks.  fixed.

Reply:
this example to be in the same PR as the guess_mime_type fix